src/client/features/enrichment/token-input.js

```diff
@@ -0,0 +1,75 @@
+//const React = require('react');
```

Comment:
I think if you turned this into it's own React component, complete with all of its own display and logic, this would be much more logical/readable.

Reply:
To clarify, I was referring to an 'gene input' component.

Reply:
Yes, this should be using a React component with a `render()` function.